ming/datastore.py

```diff
@@ -60,16 +61,19 @@ def create_datastore(uri, **kwargs):
             "Unrecognized kwarg(s) when supplying bind: %s" %
             (kwargs.keys(),))
 
-    parts = urllib.parse.urlsplit(uri)
-
-    # Create the engine (if necessary)
-    if parts.scheme:
-        if bind:
-            raise exc.MingConfigError("bind not allowed with full URI")
-        bind = create_engine(uri, **kwargs)
+    try:
+        parts = parse_uri(uri)
+    except InvalidURI:
+        urlparts = urllib.parse.urlsplit(uri)
+        parts = {"database": urlparts.path}
```

Comment:
It'd directly set the `database` variable here and in previous `try` instead of keeping ` parts` around for no other reason.